File: Temp/CPP/vital_message.cpp
#include <iostream>
#include <string>
#include <random>
#include <unistd.h>
#include <algorithm>

using std::cout;
using std::endl;
using std::string;
using std::cin;

bool playing = true;
void main_game();
string get_message(int difficulty);
int get_difficulty();
void clear_screen();
string to_upper_case(string message);

int main (int argc, char** argv) {

	string message_sent;
	char answer;

	clear_screen();

	while (playing == true) {

		cout<<"The Vital Message"<<endl;
		cout<<"-----------------"<<endl<<endl;

		int difficulty = get_difficulty();
		
		string message = get_message(difficulty);

		clear_screen();

		cout<<"Send this message"<<endl;
		cout<<message<<endl;
		sleep(5);
		clear_screen();

		cout<<"Send the message"<<endl;
		cin>>message_sent;

		message_sent = to_upper_case(message_sent);

		if (message == message_sent) {

			cout<<"The message is correct"<<endl;
			cout<<"The war is over"<<endl;
		} else {

			cout<<"You got it wrong"<<endl;
			cout<<"You should have sent: "<<message<<endl;
		}

		cout<<"Do you want to play again?"<<endl;
		cin>>answer;

		answer = toupper(answer);

		if( answer == 'N') {
			playing = false;
		}

	}

}

string to_upper_case(string message) {

	std::for_each(message.begin(), message.end(), [](char & c) {
		c = ::toupper(c);
	});

	return message;
}

string get_message(int difficulty) {

	int char_number = 65;
	string message;

	for (int x = 0; x<difficulty; x++) {

		std::random_device rd;
		std::mt19937 gen(rd());
		std::uniform_int_distribution <> distr(char_number,char_number+25);

		message += (char)distr(gen);
	}

	return message;

}

//Asks for a difficulty level
int get_difficulty() {

	int difficulty;
	bool validate = true;

	while (validate) {
		
		cout<<"How difficult (4-10)?"<<endl;
		
		//Validates the input to confirm that it is an integer
		if (cin>>difficulty) {

			if (difficulty > 3 and difficulty < 11) {

				return difficulty;
			} 
		}

		//if it is not an integer, or falls out of range
		//We clear the buffer and try again.
		cin.clear();
		cin.ignore();

		cout<<"Please enter a number between 4 and 10"<<endl;
	}

	return difficulty;

}

void clear_screen() {

	int number = system("clear");

	if (number == 0) {
		cout<<endl;
	}
}

File: Temp/vital_message.py
from os import system
import random
from time import sleep

#Clears the screen
def clear():
	_ = system('clear')

def main():

	#Sets the game loop
	game = True

	while game == True:
		print("The Vital Message")
		print("-----------------")
		print()
		print("How difficult? (4-10)")
	
		#Gets the difficulty, and length of message based on the difficulty
		difficulty = get_difficulty()
		message = get_message(difficulty)

		#Prints the message on the screen and then clears it
		clear()
		print("Send this message: \n")
		print(message)
		sleep(5)
		clear()

		print("Send the message")
		attempt = input()
		attempt = attempt.upper()

		#Checks to see if the message is correct
		if attempt == message:
			print("The message is correct")
			print("The war is over!!")
		else:
			print("You got it wrong")
			print("You should have sent: {}".format(message))

		print("Do you want to play again?")
		answer = input()

		if (answer.upper() == 'N'):
			game = False

#Function to generate the message
def get_message(difficulty):

	message = ""
	number = 65

	#Returns a string of letters based on the difficulty
	for x in range(difficulty):
		n = random.randint(number,number+26)
		letter = chr(n)
		message += letter
	return message;

#Gets the difficulty of the game, which determines the length of the message
def get_difficulty():

	valid = False;

	while valid == False:

		#Confirms that the difficulty is a number, and between 4 and 10
		try:
			difficulty = int(input())

			if (difficulty > 3) and (difficulty <11):
				return difficulty

			print("You must enter a value beteen 4 and 10")
		except:
			print("You must enter a number")

if __name__ == "__main__":
	clear()
	main()